### PR TITLE
Fix frontend uniqueness check, where IP=* is a NULL pointer

### LINKED ISSUE #317: Core dump when listening to multiple interfaces
Hello,

I am trying to get hitch listen on both ipv4 and ipv6 simultaneously. I got it listening on either ipv4 or ipv6, but when i configure it to listen on both like the following:
frontend = {
    host = "*"
    port = "443"
}

frontend = {
    host = "fd22:5979:45f3:9471::1"
    port = "443"
}

I get: /usr/sbin/hitch --pidfile=/run/hitch/hitch.pid --config=/etc/hitch/hitch.conf
Erreur de segmentation (core dumped)

or on the syslog:
 kernel: hitch[1663]: segfault at 0 ip 00007f7b0d40943e sp 00007ffd685b57d8 error 4 in libc-2.28.so[7f7b0d378000+1ba000]
hitch.service: Main process exited, code=dumped, status=11/SEGV
hitch.service: Failed with result 'core-dump'.
systemd-coredump[1666]: Process 1663 (hitch) of user 0 dumped core.#012#012Stack trace of thread 1663:#012#0  0x00007f7b0d40943e __strcmp_sse2 (libc.so.6)#012#1  0x0000559bb0cbcee4 front_arg_add (hitch)#012#2  0x0000559bb0ccf7f2 yyparse (hitch)#012#3  0x0000559bb0cbf7d0 config_parse_cli (hitch)#012#4  0x0000559bb0cb8041 main (hitch)#012#5  0x00007f7b0d39b813 __libc_start_main (libc.so.6)#012#6  0x0000559bb0cba3be _start (hitch)
kernel: hitch[1681]: segfault at 0 ip 00007fd4f1d6443e sp 00007ffd798e9378 error 4 in libc-2.28.so[7fd4f1cd3000+1ba000]
 
System: centos-release-8.0-0.1905.0.9.el8.x86_64 running as KVM virtual machine.
Hitch: hitch-1.5.0-3.el8.x86_64

Is someone able to reproduce this ?

### PR DESCRIPTION
**Pull request by dmatetelki**
Fixes #317

### CODE REVIEW

File: src/configuration.c [hunk 1]
```diff
@@ -745,16 +745,32 @@ static int
...
-			"(matching IP and port) definition: '%s:%s'.",
-			    fa->ip, fa->port);
-			return (0);
+			retval = 0;
+			break;
+		}
+		else if (cur_fa->ip == NULL || fa->ip == NULL)
+			continue;
+		else {
+			if (strcmp(cur_fa->ip, fa->ip) == 0 &&
```
Comment (Dridi) [inline src/configuration.c]: The `else { if` double block can simply be an `else if` block.

**Comment by Dridi** [comment]
What I would like is first an explanation of the root cause that would justify this fix. Why is it that we end up with a null IP?

Another question: is the test guaranteed to work everywhere? I didn't bother to check, I'm working on something else.

Otherwise I'm happy with the change.

**Comment by dmatetelki** [comment]
If the frontend definition is "bind on all IPs" with the wildcard: "*",
then the IP value is NULL: https://github.com/varnish/hitch/blob/master/src/cfg_parser.y#L160-L161

If we add multiple frontend definitions, we iterate through them, to check uniqueness.

If there is a non-last frontend definition with IP NULL, then when we compare the current frontend to it, an strcmp call could dereference a NULL pointer, `fa->ip`: https://github.com/varnish/hitch/blob/master/src/configuration.c#L749

**Comment by FathiBenNasr** [comment]
Thanks.

**Pull request merged**